### ISSUE TITLE
chef_version constraint is not accurate.

### ISSUE THREAD

**Issue by limitusus** [sous-chefs/apache2#737]
## :ghost: Brief Description

`unified_mode` method is introduced Chef 15.3 (https://docs.chef.io/release_notes_client/#custom-resource-unified-mode), so the Chef version constraint should be `>= 15.3`.
(Not critical issue, of course)

Error below:

```
================================================================================
Recipe Compile Error in /var/chef/cache/cookbooks/apache2/resources/conf.rb
================================================================================

NoMethodError
-------------
undefined method `unified_mode' for #<Class:0x000000000482f7b8>

Cookbook Trace:
---------------
  /var/chef/cache/cookbooks/apache2/resources/conf.rb:1:in `class_from_file'

Relevant File Content:
----------------------
/var/chef/cache/cookbooks/apache2/resources/conf.rb:

  1>> unified_mode true
  2:  
  3:  property :path, String,
  4:           default: lazy { "#{apache_dir}/conf-available" },
  5:           description: 'Path to the conf-available directory'
  6:  
  7:  property :root_group, String,
  8:           default: lazy { node['root_group'] },
  9:           description: ''
 10:  

System Info:
------------
chef_version=15.2.20
platform=ubuntu
platform_version=14.04
ruby=ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-linux]
program_name=/usr/bin/cinc-client
executable=/opt/cinc/bin/cinc-client
```

## :pancakes: Cookbook version

v8.1.2+

## :woman_cook:  Chef-Infra Version

Cinc-client 15.2.20

## :tophat: Platform details

Ubuntu 14.04

## Steps To Reproduce

Include apache2 in the run list (compile time issue)

## :police_car: Expected behavior

It should not proceed to compile phase; should fail in cookbook version check.

## :heavy_plus_sign: Additional context

This is introduced here https://github.com/sous-chefs/apache2/pull/689

**Comment by ramereth** [comment]
@limitusus please feel free to make a PR which fixes this.

**Comment by damacus** [comment]
Resolved via #738
